docs/extensions/readme.md

```diff
@@ -1192,6 +1193,145 @@ When set, specifies the header parameter to be used instead of `x-ms-client-requ
 }
 ```
 
+## x-ms-arm-id-details
+
+Can only be set on `"type": "string"` fields with `"format": "arm-id"`.
+
+When set, specifies the set of resource types which can be referenced by this `arm-id`. If this extension isn't provided for a particular `arm-id`, the field can refer to any valid ARM ID.
+
+**Parent element**: [Parameter Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#parameterObject), [Schema Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#schemaObject), or [Items Object](https://github.com/swagger-api/swagger-spec/blob/master/versions/2.0.md#itemsObject)
+
+### Schema
+
+| Field Name        | Type                | Description                                      |
+| ----------------- | ------------------- | ------------------------------------------------ |
+| allowedResources  | `[AllowedResource]` | **Required** An array of allowed ARM resources. Each element represents a particular type of ARM resource which can be referred to by this `arm-id`. |
+
+**AllowedResource schema**:
+
+| Field Name  | Type       | Description                                                                                                                                                                                                         |
+| ----------- | ---------- | ------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------- |
+| scopes      | `[string]` | An array of scopes. See [Allowed Scopes](#allowed-scopes). If not specified, the default scope is `["ResourceGroup"]`.                                                                                              |
+| type        | `string`   | **Required** The type of resource that is being referred to. For example `Microsoft.Network/virtualNetworks` or `Microsoft.Network/virtualNetworks/subnets`. See [Example Types](#example-types) for more examples. |
+
+#### Allowed Scopes
```

Comment:
What are the thoughts on how scopes should be interpreted?  Does this apply a restriction to the same tenant / subscription / resource group / management group as the referring resource, or something else?

I worry about making this too deployment-specific.  It would be nice to apply this in general to encode actual 'locality' restrictions that could also be used both for smart tools to validate an input reference or for for smart tools to deploy the resource.  One such restriction would be 'location', for resources that must be in the same region as the referring resource.

Reply:
So, management groups typically sit 'above' subscriptions, so, unless I am wildly mistaken, you can *only* deploy a virtual network to a resource group.  Management Groups are a good deployment location for extension resources, but don't apply to resources that are Tracked Resources or Subscription resources, or resource group resources.

I guess here, you want to be able to deploy the resource without having to know anything about the properties of the resource type, is that correct?

Reply:
> I guess here, you want to be able to deploy the resource without having to know anything about the properties of the resource type, is that correct?

Yes, exactly. Maybe I need to do some documentation improvements (or we need to pick a different name than `Scope`?). As discussed over teams the issue is that this is a reference to a say `Microsoft.Network/virtualnetworks`, and _at the reference site_ there's no knowledge of what that thing looks like. The objective for this annotation would be that client tools could give better hints/warnings about ARM references, so ideally given this annotation we can look at an ARM ID and confirm if it's valid. Without a `scope` section though we couldn't confirm that:
`/subscriptions/<id>/Microsoft.Network/virtualNetworks/myVnet` was an invalid reference, because it's not at the right scope.

Reply:
I would say, this is good for the first version, and perhaps we can augment this later with some restrictions around the resource (for example, that it has to be in the same region,